src/webview/components/memory-widget.tsx

```diff
@@ -84,9 +86,11 @@ export class MemoryWidget extends React.Component<MemoryWidgetProps, MemoryWidge
                 groupsPerRow={this.props.groupsPerRow}
                 offset={this.props.offset}
                 count={this.props.count}
+                effectiveAddressLength={this.props.effectiveAddressLength}
                 fetchMemory={this.props.fetchMemory}
                 isMemoryFetching={this.props.isMemoryFetching}
                 scrollingBehavior={this.props.scrollingBehavior}
+                addressPadding={this.props.addressPadding}
```

Comment:
I have some misgivings about the number of places we're passing around the new memoized padding / length values when they're relevant to only one column, and we should avoid knowing too much about column internals outside of those contributions.

One idea about how to handle it would be to introduce a new lifecycle on the columns so that they can update any relevant state `onComponentDidUpdate` or whatever seems best. Another (obviously beyond the scope of this PR) would be to use [something other than React](https://github.com/eclipse-cdt-cloud/vscode-memory-inspector/issues/15) so that individual columns could have tighter control of their own rendering.

Reply:
Do you mean -- conceptually -- `onComponentDidUpdate` in `App`, we invoke an update on each column, passing them a new object state to avoid having to pass column-specific state down to the `MemoryWidget`/`MemoryTable`?
That sounds like a good idea to me!
I feel though such a change probably deserves a separate PR, wdyt?

Reply:
> Do you mean -- conceptually -- onComponentDidUpdate in App, we invoke an update on each column, passing them a new object state to avoid having to pass column-specific state down to the MemoryWidget/MemoryTable?

Yes, exactly. That way columns are kept up-to-date on changes in props/state and can store the results of any calculations they'd otherwise repeat.

> I feel though such a change probably deserves a separate PR, wdyt?

I'd be inclined to do it here because this PR is introducing a new prop that's being passed in a style we might like to avoid, but if you feel strongly, we could put it off.

Reply:
To be frank: I agree with @planger that this should be addressed in a separate refactoring PR.
It is a great idea to have more fine granular control over updates. In the sense of avoiding to flood components with "unsed" or just passed through props. But it sounds like this is something that should be properly analyzed and rolled out across the board. I am sure there is more potential for this beyond the address padding. And a separate PR will give clearer separation of the feature enhancement and refactoring work.

Reply:
I'm ok with that approach, though I think it's not generally a good idea to acknowledge that something is not the direction we want to go and then move forward with it anyway. I've approved the PR - @planger, feel free to make whatever changes to the wording of the preference enum descriptions that you prefer and merge at your convenience.

Reply:
Thanks, @colin-grant-work ! I appreciate that the situation isn't ideal. Let's make sure that we address this as soon as we can.